### ISSUE TITLE
Semantic Mediawiki extension

### ISSUE THREAD

**Issue by PabloCastellano** [postmarketOS/wiki#11]
> [Semantic MediaWiki (SMW)](https://www.semantic-mediawiki.org/wiki/Semantic_MediaWiki) is an extension to MediaWiki that allows for annotating semantic data within wiki pages, thus turning a wiki that incorporates the extension into a semantic wiki. Data that has been encoded can be used in semantic searches, used for aggregation of pages, displayed in formats like maps, calendars and graphs, and exported to the outside world via formats like RDF and CSV.

Here goes a proposal for our wiki.By the way I have a proposal for the wiki. A friend of mine is really good with wikis and in a common project he did a good job adding the semantic mediawiki extension. So maybe he can help us to make our wiki semantic as well
It would allow us to do things like:

list devices with same SoC
list people with some specific devices
list devices other common things (redish screen, supporting 3d acceleration, ...)
Plus the extension also provides an HTTP API
Or e.g. you can just add the Category "supported device" and it will automatically move the device from "Working progress" table to "Supported"

**Comment by ollieparanoid** [comment]
Code is [very well maintained](https://github.com/SemanticMediaWiki/SemanticMediaWiki) (that's always important to me when choosing extensions/programs).

> Add the Category "supported device" to a device page and it will automatically move the device name from the Work in progress table to the Officially supported one.

The "officially supported" table looks different than the WIP table, it has all these yes/no values. Are you sure that we could automatically move it in that table?

>* List devices with same SoC
>* List devices other common things (redish screen, supporting 3d acceleration, ...)

That sounds useful, but at the same time we could solve this with MediaWiki's categories today.

The documentation is a bit overwhelming, I can't quickly find a "how to add a semantic tag to a wiki page" example (though I've read it for a few minutes).

My first impression is, that this is a bit overkill to be honest.

Would this improve our search? (maybe solve #9?)

**Comment by PabloCastellano** [comment]
> The "officially supported" table looks different than the WIP table, it has all these yes/no values. Are you sure that we could automatically move it in that table?

Yes! Pages can have attributes and tables can be created dynamically based on categories and attributes. In this case, attributes can be `device_supports_3d`, `device_supports_audio`, `device_status` ...

> That sounds useful, but at the same time we could solve this with MediaWiki's categories today.

Think of it as taking the pages in a category in showing them in a table with as many columns as attributes have these pages. And of course you can decide which attributes you want to show/hide in each case.

You have a working example [here](https://www.semantic-mediawiki.org/wiki/Help:Inline_queries#Parser_function_.23ask). Note that Berlin is included in that table [because](https://www.semantic-mediawiki.org/w/index.php?title=Demo:Berlin&action=edit) it has the attribute `[[located in::Germany]]`.
